src/parser/parser.py

```diff
@@ -0,0 +1,64 @@
+"""Parses a PDF document.
+"""
+import logging
+
+
+import slate
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Article:
+    """Class for storing articles.
+    """
+
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles 
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+                pdf_text[i + 1] == str(article_count) + ".-"
```

Comment:
This is a good opportunity to use regular expressions! https://www.w3schools.com/python/python_regex.asp

And also string formatting: https://pyformat.info/

If you're not familiar with regular expressions, your three `or`s and the if statement could be combined to something like: 
`re.split("art(i|e)culo {0}(-|.)-*".format(str(article_count))` I don't have an accented i on my keyboard for articulo, so I replaced it with an "e" for this example. 

Here's a rough explanation: https://regex101.com/r/N6DsXl/1/

(You just want full matches, you don't care about group matches in this case).

Reply:
Yes, I thought about using regex and @Domene99 suggested using it. I'm on it.

Reply:
With regex you can even skip the whole while loop!